pkg/proxy/iptables/proxier.go

```diff
@@ -941,8 +941,11 @@ func (proxier *Proxier) syncProxyRules() {
 		glog.Errorf("Failed to execute iptables-restore: %v", err)
 		// Revert new local ports.
 		for k, v := range newLocalPorts {
```

Comment:
It is poorly named.  It represents the set of local ports "after the current state is synced".  It's "new" as in "old set vs new set" not as in "previously not opened".

Minhan, can you

* rename this variable to something like "finalPortsMap" or "replacementPortsMap"
* change the comment on it to say "Accumulate the set of local ports that we will be holding open once this update is complete"
* add comments at L733 and L796 to the effect of "this port was open before and is still needed"
* change the comment in your fix to something like "Only close *newly opened* local ports - leave ones that were open before this update"
* change the glog.Errorf to V(2).Infof and make the message clearer.  Something like "Not holding port %s open after failure"

If we want to get really fancy, maybe we want (as a different PR) to parse the iptables error looking for the exact error message, recognize any error with a line number, and print `lines`.  Even better, make a function that prints it with line numbers.